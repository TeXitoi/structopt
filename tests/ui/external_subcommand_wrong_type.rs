use structopt::StructOpt;
use std::ffi::CString;

#[derive(StructOpt, Debug)]
struct Opt {
    #[structopt(subcommand)]
    cmd: Command,
}

#[derive(StructOpt, Debug)]
enum Command {
    #[structopt(external_subcommand)]
    Other(Vec<CString>)
}

fn main() {
    let opt = Opt::from_args();
    println!("{:?}", opt);
}