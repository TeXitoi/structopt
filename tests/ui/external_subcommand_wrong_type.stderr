error[E0308]: mismatched types
  --> $DIR/external_subcommand_wrong_type.rs:13:15
   |
13 |     Other(Vec<CString>)
   |               ^^^^^^^ expected struct `std::ffi::CString`, found struct `std::ffi::OsString`
   |
   = note: expected struct `std::vec::Vec<std::ffi::CString>`
              found struct `std::vec::Vec<std::ffi::OsString>`
